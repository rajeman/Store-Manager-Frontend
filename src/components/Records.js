import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { fetchSales } from '../actions/records';
import Record from './Record';

class Records extends React.Component {
  componentDidMount() {
    this.props.dispatch(fetchSales())
  }
  render() {
    const { sales } = this.props.sales;
    return (
      <div className="container">
        <div className="wrapper">
          <table id="orders-table">
            <tr>
              <th>Attendant ID</th>
              <th>Attendant Name</th>
              <th>Order Id</th>
              <th>Time Created</th>
              <th>Total Item</th>
              <th>Order Price</th>
            </tr>
            {sales.map((sale) =>
              <Record sale={sale} key={sale.order_id} ></Record>)
            }
          </table>
        </div>
      </div>
    );
  }
}
const mapStateToProps = (state) => state;
export default connect(mapStateToProps)(Records);



