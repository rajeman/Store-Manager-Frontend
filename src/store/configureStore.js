import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import productReducer from '../reducers/products';
import authReducer from '../reducers/auth';
import saleReducer from '../reducers/records';
import navigationReducer from '../reducers/navigation';
import attendantReducer from '../reducers/attendants';
import cartReducer from '../reducers/cart';

export default () => createStore(
  combineReducers({
    products: productReducer,
    auth: authReducer,
    navigation: navigationReducer,
    sales: saleReducer,
    attendants: attendantReducer,
    cart: cartReducer,
  }), applyMiddleware(thunk),
);
