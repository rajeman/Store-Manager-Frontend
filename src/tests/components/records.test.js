import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import axios from 'axios';
import Records from '../../components/Records';

jest.mock('axios');

describe('RECORDS TEST SUITE', () => {
  it('should render the record component', () => {
    const response = {
      data:
      {
        message: 'successfully fetched user info',
      }
    };
    axios.get.mockResolvedValue(response);
    const initialState = {
      sales: {
          sales: [{order_id: 25}]
      }
    };
    const mockStore = configureStore([thunk]);
    const store = mockStore(initialState);
    const component = mount(
      <Provider store={store}>
        <Records />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
