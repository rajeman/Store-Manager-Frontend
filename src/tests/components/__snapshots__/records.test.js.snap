// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RECORDS TEST SUITE should render the record component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Records)>
    <Records
      dispatch={[Function]}
      sales={
        Object {
          "sales": Array [
            Object {
              "order_id": 25,
            },
          ],
        }
      }
    >
      <div
        class="container"
      >
        <div
          class="wrapper"
        >
          <table
            id="orders-table"
          >
            <tr>
              <th>
                Attendant ID
              </th>
              <th>
                Attendant Name
              </th>
              <th>
                Order Id
              </th>
              <th>
                Time Created
              </th>
              <th>
                Total Item
              </th>
              <th>
                Order Price
              </th>
            </tr>
            <Record
              key="25"
              sale={
                Object {
                  "order_id": 25,
                }
              }
            >
              <tr
                onClick={[Function]}
              >
                <td />
                <td />
                <td>
                  25
                </td>
                <td>
                  undefined-NaN-NaN NaN:aN
                </td>
                <td />
                <td />
              </tr>
            </Record>
          </table>
        </div>
      </div>
    </Records>
  </Connect(Records)>
</Provider>
`;
